Isu/Services/IsuService.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.Intrinsics.X86;
+using Isu.Models;
+using Isu.Tools;
+
+namespace Isu.Services
+{
+    public class IsuService : IIsuService
+        {
+            private readonly List<Group> _groups = new ();
+
+            public Group AddGroup(string name)
+            {
+                int n;
+                string sub = name.Substring(2);
+                if (name.Length != 5 || !name.StartsWith("M3") || !int.TryParse(sub, out n))
```

Comment:
int n; c 17-ой строчки можно заинлайнить
```suggestion
                if (name.Length != 5 || !name.StartsWith("M3") || !int.TryParse(sub, out var n))
```